core/src/cache.rs

```diff
@@ -483,9 +486,39 @@ impl Cache {
             InputFormat::Json => serde_json::from_str(self.files.source(file_id))
                 .map(|t| (attach_pos(t), ParseErrors::default()))
                 .map_err(|err| ParseError::from_serde_json(err, file_id, &self.files)),
-            InputFormat::Yaml => serde_yaml::from_str(self.files.source(file_id))
-                .map(|t| (attach_pos(t), ParseErrors::default()))
-                .map_err(|err| (ParseError::from_serde_yaml(err, file_id))),
+            InputFormat::Yaml => {
+                // YAML files can contain multiple documents. If there is only
+                // one we transparently deserialize it. If there are multiple,
+                // we deserialize the file as an array.
+                let de = serde_yaml::Deserializer::from_str(self.files.source(file_id));
+                let mut terms = de
+                    .map(|de| {
+                        RichTerm::deserialize(de)
+                            .map(attach_pos)
```

Comment:
It would make more sense to attach the sub-span corresponding to the single document, I guess. Doesn't seem like this can be easily extracted from `serde_yaml`, though...

Reply:
Yeah, I guess that's also the reason that the subterms of a term imported from something other than Nickel don't carry positions. I wonder if that breaks some assumptions in the evaluator...